src/lightning_app/core/queues.py

```diff
@@ -291,3 +301,98 @@ def is_running(self) -> bool:
             return self.redis.ping()
         except redis.exceptions.ConnectionError:
             return False
+
+
+class HTTPQueue(BaseQueue):
+    def __init__(self, name: str, default_timeout: float):
+        """
+        Parameters
+        ----------
+        name:
+            The name of the Queue to use. In the current implementation, we expect the name to be of the format
+            `appID_queueName`. Based on this assumption, we try to fetch the app id and the queue name by splitting
+            the `name` argument.
+        default_timeout:
+            Default timeout for redis read
+        """
+        if name is None:
+            raise ValueError("You must specify a name for the queue")
+        self.app_id, self.name = self._split_app_id_and_queue_name(name)
+        self._original_name = name  # keeping the name for debugging
+        self.default_timeout = default_timeout
+        self.client = HTTPClient(base_url=HTTP_QUEUE_URL, log_callback=debug_log_callback)
+
+    def get(self, timeout: int = None) -> Any:
+        if not self.app_id:
+            raise ValueError(f"App ID couldn't be extracted from the queue name: {self._original_name}")
+
+        # it's a blocking call, we need to loop and call the backend to mimic this behavior
+        if timeout is None:
+            while True:
+                try:
+                    return self._get()
+                except queue.Empty:
+                    time.sleep(HTTP_QUEUE_REFRESH_INTERVAL)
+
+        # make one request and return the result
+        if timeout == 0:
+            return self._get()
```

Comment:
Should this one be wrapped into a try catch, what happened if the service doesn't answer ?

Reply:
We are implicitly retrying for 2 minutes for most of the failure modes (except the service completely goes down or if we get a connection refused error). I have a TODO somewhere in the code to address this in the future (before making HTTPQueue the default Queue)

What's the behaviour you'd expect if the service is down? We'll retry forever, in a defined frequency?

Reply:
I am going to keep this for a future PR